app/models/manageiq/providers/google/cloud_manager/provision_workflow.rb

```diff
@@ -0,0 +1,23 @@
+class ManageIQ::Providers::Google::CloudManager::ProvisionWorkflow < ::MiqProvisionCloudWorkflow
+  def allowed_instance_types(_options = {})
+    source = load_ar_obj(get_source_vm)
+    ems = get_targets_for_ems(source, :cloud_filter, Flavor, 'flavors')
+    ems.each_with_object({}) { |f, h| h[f.id] = display_name_for_name_description(f) }
+  end
+
+  def allowed_disk_sizes(_options = {})
+    min = 10
+    max = 10 * 1024 # 10 TB
+    (min..max).each_with_object({}) { |i, h| h[i] = i.to_s }
```

Comment:
Not sure this is what you want to do as it will create a drop-down with over 10,000 items.  For starters, I would suggest not using this method and just add a few items directly in the dialog the way it is done for the field `:retirement` in the dialog in this PR.  For infrastructure provisioning we do the same thing for memory size and users can adjust the values by changing the dialog.  Would this fit your needs?